.github/workflows/pull-request-checks.yaml

```diff
@@ -11,14 +11,84 @@ on:
       - master
 
 jobs:
-  build:
+  check_version:
+    name: Ensure component commits modify component versions
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2.4.0
+      name: Checkout repo
+      with:
+        # See https://github.com/actions/checkout#checkout-v2
+        # This will be slow. The intent is to fetch all commits
+        # since the merge-base (the commit where we branched off)
+        # so we can check the git diff against all changed files.
+        # By default, the checkout action only returns the last commit,
+        # There's no native way to do this in the checkout action, so 
+        # we have to fetch the entire history. See
+        # https://github.com/actions/checkout/issues/266#issuecomment-638346893
+        fetch-depth: 0
+    - uses: jitterbit/get-changed-files@v1
+      id: changed_files
+      name: Get changed files
+    # We need to get a diff of the base and HEAD commit below
+    - name: Return base commit
+      id: base_commit
+      uses: actions/github-script@v6
+      with:
+        script: return context.payload.pull_request?.base?.sha
+        result-encoding: string
+    - name: Return HEAD commit
+      id: head_commit
+      uses: actions/github-script@v6
+      with:
+        script: return context.payload.pull_request?.head?.sha
+        result-encoding: string
```

Comment:
**-Minor/Comment-**
I think we could get the base_commit and head_commit from the `github` [context](https://docs.github.com/en/actions/learn-github-actions/contexts) using an [expression](https://docs.github.com/en/actions/learn-github-actions/expressions) instead.

E.g., using the [`env` context](https://docs.github.com/en/actions/learn-github-actions/contexts#env-context):
```sh
    - id: git_diff_on_components
      name: Check git diff for version changes
      env:
        head_commit: ${{ github.event.pull_request.head.sha }}
        base_commit: ${{ github.event.pull_request.base.sha }}
      run: |-
        commit_contained_components_that_did_not_modify_version=0
        for f in ${{ steps.changed_files.outputs.all }}
        do
          ext="${f##*.}"
          # Only run this check on modified sources or actions, excluding common files
          if ([[ "$ext" == "js" ]] || [[ "$ext" == "mjs" ]]) && \
             ([[ "${f}" == *components/**/sources/* ]] || [[ "${f}" == *components/**/actions/* ]]) && \
             [[ "${f}" != *common*.*js ]] && [[ "${f}" != **/actions/common/* ]] && [[ "${f}" != **/sources/common/* ]] && \
             [[ "$(<${f})" == *"version:"* ]]
          then
            BASE_COMMIT=${{env.base_commit}}
            HEAD_COMMIT=${{env.head_commit}}
```